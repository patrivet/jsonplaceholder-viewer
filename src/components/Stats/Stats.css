.stats {
  position: sticky;
  top: 0;
  z-index: 1;
  background-color: white;
  display: flex;
  text-align: center;

  display: flex;
  justify-content: center;
}
