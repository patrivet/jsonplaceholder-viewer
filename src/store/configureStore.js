import { createStore, compose, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import rootReducer from "./index";

export default function configureStore(initialState) {
  return createStore(
    rootReducer,
    initialState,
    compose(applyMiddleware(thunk, logger))
  );
}
