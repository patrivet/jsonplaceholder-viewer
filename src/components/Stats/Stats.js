import { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import "./Stats.css";

// Custom components
import { getPostStats } from "../../utils/helperFunctions";

const Stats = () => {
  const useStyles = makeStyles({
    root: {
      width: "30%",
      margin: 20,
    },
    statsValue: {
      fontWeight: 600,
    },
  });

  const [totalWords, setTotalWords] = useState(0);
  const [mostFrequentWords, setMostFrequentWords] = useState([]);

  const posts = useSelector((state) => state.posts.list);
  const loading = useSelector((state) => state.posts.loading);

  useEffect(() => {
    // Only compute when not loading & posts to process
    if (!loading && posts) {
      const { totalWords, mostFrequentWords } = getPostStats(posts);
      setTotalWords(totalWords);
      setMostFrequentWords(mostFrequentWords);
    }
  }, [posts, loading]);

  const classes = useStyles();

  return (
    <div className="stats">
      <Card className={classes.root}>
        <CardContent>
          <Typography variant="h5" component="h2">
            Total Word Count:
          </Typography>
          <Typography
            variant="h5"
            component="h2"
            className={classes.statsValue}
          >
            {totalWords}
          </Typography>
        </CardContent>
      </Card>
      <Card className={classes.root}>
        <CardContent>
          <Typography variant="h5" component="h2">
            Top 5 Most Frequent Words:
          </Typography>
          <Typography
            variant="h5"
            component="h2"
            className={classes.statsValue}
          >
            {mostFrequentWords.toString()}
          </Typography>
        </CardContent>
      </Card>
    </div>
  );
};

export default Stats;
