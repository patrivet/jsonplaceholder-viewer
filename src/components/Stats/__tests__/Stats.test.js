import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Stats from "../Stats";
describe("Stats", () => {
  const mockStore = configureStore();
  let store;
  test("renders correctly", () => {
    store = mockStore({
      posts: [],
      loading: false,
    });

    const { container } = render(
      <Provider store={store}>
        <Stats />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
