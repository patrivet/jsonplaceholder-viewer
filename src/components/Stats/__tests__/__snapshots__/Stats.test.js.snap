// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats renders correctly 1`] = `
<div>
  <div
    class="stats"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5"
        >
          Total Word Count:
        </h2>
        <h2
          class="MuiTypography-root makeStyles-statsValue-2 MuiTypography-h5"
        >
          0
        </h2>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5"
        >
          Top 5 Most Frequent Words:
        </h2>
        <h2
          class="MuiTypography-root makeStyles-statsValue-2 MuiTypography-h5"
        />
      </div>
    </div>
  </div>
</div>
`;
